### PR TITLE
test(icrc_ledger): FI-1400: Add golden state ledger verification with ckBTC and ckETH workarounds

### PR DESCRIPTION
**Pull request by mbjorkqvist**
Add ledger state verification wrt. account balances and allowances for the ck-ledgers. The ckBTC and ckETH ledgers require some workarounds due to some burns not including the account of the minter in the spender field.

### CODE REVIEW

File: rs/rosetta-api/icrc1/ledger/sm-tests/src/in_memory_ledger.rs [hunk 1]
```diff
@@ -69,6 +70,7 @@ where
     pub allowances: HashMap<K, Allowance<Tokens>>,
     pub total_supply: Tokens,
     pub fee_collector: Option<AccountId>,
+    pub burns_without_spender: Option<BurnsWithoutSpender<AccountId>>,
```
Comment (maciejdfinity) [inline rs/rosetta-api/icrc1/ledger/sm-tests/src/in_memory_ledger.rs:73]: Does this variable (and above) need to be public? In another PR Nikolas mentioned we use such practice, but I am not sure what is the benefit? I am not against making it public if needed, but what is the benefit of using pub by default?

Comment (mbjorkqvist) [reply rs/rosetta-api/icrc1/ledger/sm-tests/src/in_memory_ledger.rs]: Good point; I also prefer to have fields, methods etc. private if it's not necessary to have them public. I changed these to private.

File: .github/workflows-source/schedule-daily.yml [hunk 2]
```diff
@@ -135,7 +135,7 @@ jobs:
   fi-tests-nightly:
     name: Bazel Test FI Nightly
     <<: *dind-large-setup
-    timeout-minutes: 120
+    timeout-minutes: 240
```
Comment (mbjorkqvist) [inline .github/workflows-source/schedule-daily.yml:138]: With these changes, the FI nightly tests timed out after 2h, so bumping this to 4h. Will consider options before merging if the nightly test passes with this setting.

Comment (mbjorkqvist) [reply .github/workflows-source/schedule-daily.yml:138]: [This more recent run](https://github.com/dfinity/ic/actions/runs/10610013902) completed in 73min, so it looks like it is quite dependent on the busyness of the hardware it is running on? I'd still leave in the longer timeout for now.

**Review by maciejdfinity — APPROVED** [review]

**Review by marko-k0 — APPROVED** [review]

**Pull request merged**